src/Event/Event.jl

```diff
@@ -1,6 +1,36 @@
 """
-    mutable struct Event{T <: SSDFloat}
+  to_internal_point(pt::AbstractCoordinatePoint)
+        
+Converts any `AbstractCoordinatePoint` (Cartesian or Cylindrical, with or without units) to a `CartesianPoint`.
+"""
+function to_internal_point(pt::CartesianPoint)
+    T = pt.x isa Unitful.Quantity ? float(typeof(ustrip(pt.x))) : typeof(pt.x)
+    x = pt.x isa Unitful.Quantity ? float(to_internal_units(pt.x)) : pt.x
+    y = pt.y isa Unitful.Quantity ? float(to_internal_units(pt.y)) : pt.y
+    z = pt.z isa Unitful.Quantity ? float(to_internal_units(pt.z)) : pt.z
+    return CartesianPoint{T}(x, y, z)
+end
+
+function to_internal_point(pt::CylindricalPoint)
+    T = pt.r isa Unitful.Quantity ? float(typeof(ustrip(pt.r))) : float(typeof(pt.r))
+    r = pt.r isa Unitful.Quantity ? float(to_internal_units(pt.r)) : float(pt.r)
+    φ = pt.φ isa Unitful.Quantity ? float(ustrip(uconvert(u"rad", pt.φ))) : float(pt.φ)
+    z = pt.z isa Unitful.Quantity ? float(to_internal_units(pt.z)) : float(pt.z)
+    return CartesianPoint{T}(r * cos(φ), r * sin(φ), z)
+end
+
+function to_internal_point(pt::AbstractCoordinatePoint)
+    error("Unsupported point type $(typeof(pt)). Expected CartesianPoint or CylindricalPoint.")
+end
+
+function to_internal_point(p::SVector{3, <:Unitful.Quantity})
+    vals = ustrip.(uconvert.(u"m", p))   # convert to meters and strip units
+    return SVector{3, typeof(vals[1])}(vals)  # preserves type
```

Comment:
Do we need to explicitly say this or can this just be:

```suggestion
    return SVector{3}(vals)  # preserves type
```